src/posts/dto/create-post.dto.ts

```diff
@@ -1,5 +1,11 @@
+import { IsBoolean, IsIn } from 'class-validator';
+
 export class CreatePostDto {
-  caption: string;
-  type: string;
+  caption?: string;
+
+  @IsIn(['text_poll', 'image_poll', 'mini_survey'])
```

Comment:
I think it would be better to retrieve the applicable posts type from a certain place/function instead of manually typing them and adding a new type if we had a new type to be added, and also to have a single point of truth regarding all the types that are available as If we needed to use them in another place, i.e. another function/service.

Reply:
I agree with you. However, I don't know where to put it know, So, I proceeded till I have a big picture for the project and then decide where to put it. 
Nevertheless, If you have a clue, let me know.

Reply:
why do you even restrict it? this could limit the frontend when creating a new type of posts. I remember we agreed to keep it open to the frontend to creat post types.

I would suggest to keep it open for now. let's see how uncontrolled post types could harm us.

Reply:
We agreed to let front-end have the full control of setting a post type & presenting a post based on its type. I added it for validation. If a new type is needed, they could open an issue. However, You might have different point of view so, we can leave it.

Reply:
I understand your intention. 
if adding new type of posts will only require a change to this set, then I would say it is ok to have this validation.
I am afraid that you keep building more and more features based on the post types.

Reply:
I think putting/relating it as utility function is logical acceptable, because its a helper function that (at anytime) will always have all the types of posts that we serve, so we can use it anywhere when needed. What do you think?
I'm asking for your opinions because maybe I'm wrong so we would find a better alternative.

Reply:
Removed the different types